keps/sig-node/20190925-enable-seccomp-by-default.md

```diff
@@ -0,0 +1,327 @@
+
+The performance tests were done on an Ubuntu machine, and involved an application calling a syscall a few million times. The log entries each test execution had 1 summary entry showing the amount of messages suppressed and 10 audit entries:
+
+```
+kauditd_printk_skb: 2336845 callbacks suppressed
+audit: type=1326 audit(1569825222.987:120145027): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=6974 comm="allowedsyscall" exe="/allowedsyscall" sig=0 arch=c000003e syscall=262 compat=0 ip=0x47c29a code=0x7ffc0000
+audit: type=1326 audit(1569825222.991:120145028): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=6974 comm="allowedsyscall" exe="/allowedsyscall" sig=0 arch=c000003e syscall=262 compat=0 ip=0x47c29a code=0x7ffc0000
+...
+audit: type=1326 audit(1569825222.991:120145035): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=6974 comm="allowedsyscall" exe="/allowedsyscall" sig=0 arch=c000003e syscall=262 compat=0 ip=0x47c29a code=0x7ffc0000
+audit: type=1326 audit(1569825222.991:120145036): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=6974 comm="allowedsyscall" exe="/allowedsyscall" sig=0 arch=c000003e syscall=262 compat=0 ip=0x47c29a code=0x7ffc0000
+
+```
+
+The throttling is based on the runtime kernel parameters below:
+
+```
+net.core.message_burst = 10
+net.core.message_cost = 5
+```
+
+Users can adjust the verbosity of their logs on demand, although the default settings are quite safe.
+
+
+#### Cost vs Benefit
+
+The performance overhead of auditing syscalls that not whitelisted are negligible, once the cost of enabling seccomp is taken into account. 
+
+Kubernetes should take the same approach taken by docker of enabling both AppArmor and Seccomp by default. This would make any overhead added by such features part of the cost for running secure container workloads. 
+
+Providing users with sensible default security settings outweights the performance impact incurred, specially when users are given the ability to opt-out if their prefer to do so.
+
+
+### User Stories
+
+#### Story 1
+As a user and administrator, I want to be able to assess what pods within my cluster would be affected by applying more restrictive seccomp profiles.
+
+#### Story 2
+As a user, I want to be able to assess whether my pods will be affected by more restrictive seccomp profiles.
+
+#### Story 3
+As a user and administrator, I want Kubernetes to audit all potentially dangerous system calls from my containers by default.
+
+
+
+## Implementation Details
+
+### 1. Audit mode support (Details)
+
+Kubernetes will continue to be unaware of downstream support. If a user tries to apply a seccomp profile containing unsupported actions (i.e. `SCMP_ACT_LOG`) today, the lower level dependencies (i.e. runC) will return an error. As a result, the container won't be able to start. We are proposing no changes to this behaviour.
```

Comment:
Won't Kubernetes be the one specifying the default profile? Wouldn't this break all containers in this case?